compare/compare.c

```diff
@@ -593,14 +609,12 @@ int main(int argc, char *argv[])
     ItemList list = {0};
     int result = R_ERROR;
 
-    int scale = 1;
+    int scale_x = 1, scale_y = 1;
     if (pos[SCALE]) {
-        const char *arg = argv[pos[SCALE]];
-        if (arg[0] < '1' || arg[0] > '8' || arg[1]) {
+        if (!parse_scale(argv[pos[SCALE]], &scale_x, &scale_y)) {
```

Comment:
Looks like it's possible to merge `if`s too.

Reply:
Merged.